aerleon/lib/naming.py

```diff
@@ -538,6 +542,29 @@ def GetServiceByProto(self, query: str, proto: str) -> List[str]:
                     services_set.add(parts[0])
         return sorted(services_set)
 
+    def GetFQDN(self, query: str) -> List[str]:
+        returnlist = []
+        data = query.split('#')
+        token = data[0].split()[0]
+        if token not in self.fqdn:
+            raise UndefinedAddressError(f'UNDEFINED: {token}')
+        for i in self.fqdn[token].items:
+            comment = ''
+            if i.find('#') > -1:
+                (name, comment) = i.split('#', 1)
+            else:
+                name = i
+
+            name = name.strip()
+            # Check if hostname is actually a subtoken
+            if self.token_re.match(name):
```

Comment:
Thinking out loud here: is there any possible overlap between token_re and fqdn_re?
```
(^[-_A-Z0-9]+$)
```
vs
```
^(?!:\/\/)(?=.{1,255}$)((.{1,63}\.){1,127}(?![0-9]*$)[a-z0-9-]+\.?)$
```

Reply:
Seems the answer is no: periods cannot appear in token names but are required for FQDN names.